### PR TITLE
Issue/7698

### PR DESCRIPTION
**Pull request by mikereiddigital**
Adds support for signed commits using the planetscale/ghcommit-actions github actions plus additional minor changes to support this addition.

## A reference to the issue / Description of it

#7689

## How does this PR fix the problem?

This removes the existing git commands that commit changes and adds the use of planetscale/ghcommit-actions to create a signed commit that will be accepted by the repository with the mandatory signed commits branch protection. This approach avoids the needs to store and maintain GPG keys.

## How has this been tested?

Please describe the tests that you ran and provide instructions to reproduce.

This workflow has been tested (using some added minor formatting issues in a terraform file) as shown here - https://github.com/ministryofjustice/modernisation-platform/actions/runs/10613114434. All the changes work although because it is not running against the main branch the final (unchanged) part where the PR is created fails. 

## Deployment Plan / Instructions

Will this deployment impact the platform and / or services on it?

This change amends a workflow that is run daily each weekday.

## Checklist (check `x` in `[ ]` of list items)

- [x] I have performed a self-review of my own code
- [x] All checks have passed
- [ ] I have made corresponding changes to the documentation
- [ ] Plan and discussed how it should be deployed to PROD (If needed)

## Additional comments (if any)

{Please write here}

### CODE REVIEW

File: .github/workflows/format-code.yml [hunk 1]
```diff
@@ -58,21 +58,22 @@ jobs:
...
           fi
-
-      - name: Push changes
-        if: env.changes == 'true'
-        run: |
-          git config --global push.autoSetupRemote true
-          git push
+  
+      - name: Generate signed commit for any changes that may exist
+        uses: planetscale/ghcommit-action@c7915d6c18d5ce4eb42b0eff3f10a29fe0766e4c #v1.44
```
Comment (ewastempel) [inline .github/workflows/format-code.yml:70]: We should be careful when using external actions from 3rd parties. Is it too difficult to achieve the same without using a GH action? Have you done some research on the action provider?

Comment (mikereiddigital) [reply .github/workflows/format-code.yml:70]: The above provider is one that was recommended by @jacobwoffenden in Analytics Platforms and is used by them for the same purpose. Without this, the other options would be to 
- Store & maintain our own GPG keys (which is an administrative burden for all of our repositories),
- Use the GraphQL API which is significantly more work involved. 
- The other option would be to temporarily disable signed commits for the workflow but I'm not in favour of that.

In short there is no simple or straight forward solution that I have been able to determine through this ticket.

**Review by ewastempel — APPROVED** [review]
LGTM, but it's introducing a new GH provider, so let's keep an eye on the action behaviour.
